### DISCUSSION TITLE
How to deploy the contracts?

### DISCUSSION THREAD

**Discussion by Kaleb47** [ProjectOpenSea/seaport#604] [Q&A]
### Description

Hello, while I have used hardhat before, the file architecture is not your typical hardhat project. It seems that the deploy script for all the contracts is missing. would I have to use foundry to deploy the contracts?

Thanks in advance

**Comment by ryanio** [comment]
Hey @Kaleb47, you can find the deployment instructions here: https://github.com/ProjectOpenSea/seaport/blob/main/docs/Deployment.md

**Comment by lidaamber** [comment]
Hi @ryanio is there any advantage in deploying the contracts through the Factory contract comparing to deploying directly from Ethereum address? 
If e.g. I would like to deploy my standalone test seaport, would I really need the deploying from Factory? Thanks

**Reply by ryanio** [reply]
@lidaamber if it's just for your own testing purposes, then it would make no difference if you would like to deploy your own, but we have deployed on most production and testnets so if you're not changing the code you can just use those addresses (in the main README under Deployments)
